File: api/routes/memory.py
"""
Distributed Agentic Reasoning Framework (DARF)

Memory Route
"""

from __future__ import annotations

from fastapi import (
    APIRouter,
    Depends,
)

from agents.agent_context import AgentContext

from api.dependencies import (
    get_agent_context,
    get_memory_agent,
)

from api.models.response import (
    MemoryResponse,
)

router = APIRouter(
    prefix="/memory",
    tags=["Memory"],
)


# ============================================================
# GET /memory
# ============================================================

@router.get(
    "",
    response_model=MemoryResponse,
)
def get_memory(
    context: AgentContext = Depends(
        get_agent_context,
    ),
    memory = Depends(
        get_memory_agent,
    ),
):
    
    memory_size = memory.memory.faiss_store.size()

    return MemoryResponse(
        success=True,
        message="Memory status retrieved successfully.",
        status="Active",
        vector_store="FAISS",
        embedding_model="all-MiniLM-L6-v2",
        top_k=5,
        memory_size=memory_size,
        metrics={
            "memory_size": memory.memory.faiss_store.size(),
            "top_k": memory.memory.top_k,
            "memory_health": memory.memory.health_check(),
            "working_memory": len(memory.memory.working.to_dict().get("keys", [])),
            "semantic_memory": memory.memory.semantic.count(),
            "episodic_memory": memory.memory.episodic.count(),
            "context_variables": context.variable_count(),
            "context_outputs": context.output_count(),
        },
        context=context.to_dict(),
    )

File: frontend/src/services/memory.ts
import api from "../api/client";

export interface MemoryMetrics {

    variables: number;

    outputs: number;

}

export interface MemoryContext {

    request_id: string | null;

    session_id: string | null;

    execution_id: string | null;

    current_agent: string | null;

    variables: Record<string, unknown>;

    outputs: Record<string, unknown>;

    metadata: Record<string, unknown>;

    version: string;

}

export interface MemoryResponse {

    success: boolean;

    message: string;

    status: string;

    vector_store: string;

    embedding_model: string;

    top_k: number;

    memory_size: number;

    metrics: MemoryMetrics;

    context: MemoryContext;

}

export async function getMemory(): Promise<MemoryResponse> {

    const response = await api.get("/memory");

    return response.data;

}